src/MCSolver.cpp

```diff
@@ -0,0 +1,545 @@
+
+                // sample whether fission event occurs
+                int fission_occurs =
+                    neutron.arand() < cell_mat->getSigmaFByGroup(group+1)
+                    / sigma_a;
+
+                // fission event
+                if (fission_occurs == 1) {
+
+                    // sample number of neutrons released during fission
+                    double nu = cell_mat->getNuSigmaFByGroup(1)
+                        / cell_mat->getSigmaFByGroup(1);
+                    int lower = (int) nu;
+                    int add = (int) (neutron.arand() < nu -lower);
+                    int num_neutrons = lower + add;
+                    for (int i=0; i<num_neutrons; ++i) {
+                        fission_banks->add(neutron_position);
+                        tallies[FISSIONS] += 1;
+                    }
+                }
+
+                // end neutron history
+                neutron.kill();
+
+            } // absorption event
+        } // if neutron alive
+    } // while neutron alive
+    
+    delete neutron_coord_position;
+
+    // tally crow distance
+    double crow_distance;
+    crow_distance = neutron.getDistance(neutron_position);
+    tallies[CROWS] += crow_distance;
+    tallies[NUM_CROWS] += 1;
+}
+
+
+/*
+ @brief     returns k, the ration of fission events to absoprtions plus leakages
+ @return    the effective k-value
+*/
+FP_PRECISION MCSolver::getKeff() {
+    return _k_eff;
+}
+
+/*
+ @brief     returns the geometry used by the solver
+ @return    the geometry used by the solver
+*/
+Geometry* MCSolver::getGeometry() {
+    return _geometry;
+}
+
+/*
+ @brief     returns the flux value for neutrons of a given group in a fsr
+ @param     fsr_id the id number of the fsr for which to find the flux
+ @param     group the energy group for which to find the flux
+ @return    the neutron flux
+*/
+FP_PRECISION MCSolver::getFlux(int fsr_id, int group) {
+    return _scalar_flux(_geometry->getNumFSRs(), _num_groups);
+}
+
+// functions that allow MCSolver to be compativle with Solver
+void MCSolver::computeFSRFissionRates(
+        double* fission_rates, int num_FSRs) {}
+void MCSolver::getFluxes(FP_PRECISION* out_fluxes, int num_fluxes) {}
+void MCSolver::setFluxes(FP_PRECISION* in_fluxes, int num_fluxes) {}
+void MCSolver::flattenFSRFluxes(FP_PRECISION value) {}
+void MCSolver::computeFSRFissionSources() {}
+void MCSolver::computeFSRScatterSources() {}
+void MCSolver::initializeSourceArrays() {}
+void MCSolver::addSourceToScalarFlux() {}
+void MCSolver::initializeFluxArrays() {}
+void MCSolver::computeFSRSources() {}
+void MCSolver::normalizeFluxes() {}
+void MCSolver::zeroTrackFluxes() {}
+void MCSolver::transportSweep() {}
+void MCSolver::storeFSRFluxes() {}
+void MCSolver::computeKeff() {}
+double MCSolver::computeResidual(residualType res_type){
+    return 0;
+}
+
+/*
+ @brief initialize the fsrs
+*/
+void MCSolver::initializeFSRs(Lattice* lattice) {
```

Comment:
I totally forgot about this....hmmm...we'll need to make this more general. I'll take some time to think about this.